src/main/java/WayofTime/bloodmagic/altar/BloodAltar.java

```diff
@@ -239,8 +242,14 @@ public void update() {
             tileAltar.getWorld().notifyBlockUpdate(tileAltar.getPos(), tileAltar.getWorld().getBlockState(tileAltar.getPos()), tileAltar.getWorld().getBlockState(tileAltar.getPos()), 3);
         }
 
-        if (internalCounter % 100 == 0 && (this.isActive || this.cooldownAfterCrafting <= 0))
+        if (internalCounter % 100 == 0 && (this.isActive || this.cooldownAfterCrafting <= 0)) {
+            /* Redstone Lamp below altar: Switch Off */
+            if (world.getBlockState(pos).getValue(BlockAltar.POWERED)) {
```

Comment:
Store this state lookup for later uses. Don't needlessly query the world multiple times.